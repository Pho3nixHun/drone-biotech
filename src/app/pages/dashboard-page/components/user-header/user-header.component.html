<header class="flex flex-row items-center space-x-3">
    <ng-content select="span" />
    <div class="w-full flex flex-row justify-between space-x-10">
        <div class="flex flex-col">
            <ng-content select="[h2, p]" />
        </div>
        <div class="flex flex-col justify-end">
            <ng-content select="[small, time]" />
        </div>
    </div>
</header>
