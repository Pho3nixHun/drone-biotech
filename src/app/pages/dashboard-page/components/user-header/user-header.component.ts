import { Component } from '@angular/core';

@Component({
    selector: 'app-user-header',
    imports: [],
    templateUrl: './user-header.component.html',
})
export class UserHeaderComponent {}
