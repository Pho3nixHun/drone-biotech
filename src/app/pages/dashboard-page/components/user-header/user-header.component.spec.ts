import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { UserHeaderComponent } from './user-header.component';

@Component({
    imports: [UserHeaderComponent],
    template: `
        <app-user-header>
            <h2>Should be projected</h2>
            <p>Should be projected</p>
            <small>Should be projected</small>
            <time>Should be projected</time>
            <div>Should not be projected</div>
        </app-user-header>
    `,
})
class TestHostComponent {}

describe('UserHeaderComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
