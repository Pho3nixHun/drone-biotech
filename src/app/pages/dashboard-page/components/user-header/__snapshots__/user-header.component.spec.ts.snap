// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeaderComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-user-header>
    <header
      class="flex flex-row items-center space-x-3"
    >
      <div
        class="w-full flex flex-row justify-between space-x-10"
      >
        <div
          class="flex flex-col"
        >
          <h2>
            Should be projected
          </h2>
          <p>
            Should be projected
          </p>
        </div>
        <div
          class="flex flex-col justify-end"
        >
          <small>
            Should be projected
          </small>
          <time>
            Should be projected
          </time>
        </div>
      </div>
    </header>
  </app-user-header>
</div>
`;
